lib/newrelic_httprb/instrumentation.rb

```diff
@@ -0,0 +1,33 @@
+DependencyDetection.defer do
+  named :http_rb
+
+  depends_on do
+    defined?(HTTP) && defined?(HTTP::Client)
+  end
+
+  executes do
+    ::NewRelic::Agent.logger.info 'Installing http.rb instrumentation'
+    require 'new_relic/agent/cross_app_tracing'
+    require 'newrelic_http/wrappers'
+  end
+
+  executes do
+    class HTTP::Client
+      def perform_with_newrelic_trace(request, options)
+        wrapped_request = ::NewRelicHTTP::HTTPRequest.new(request)
+
+        ::NewRelic::Agent::CrossAppTracing.tl_trace_http_request(wrapped_request) do
+          # RUBY-1244 Disable further tracing in request to avoid double
+          # counting if connection wasn't started (which calls request again).
```

Comment:
I believe the `http_client_test_cases` is a generic suite of tests for the several client instrumentations, as hinted here:

https://github.com/newrelic/rpm/blob/master/test/multiverse/suites/typhoeus/typhoeus_test.rb#L7

However, I wasn't able to require the test cases from within my gem, so I am not really sure `http.rb` is subject to the same issue worked around in `RUBY-1244`. Any thoughts on how to proceed on this one?